### ISSUE TITLE
Conflicts should be resolved when possible

### ISSUE THREAD

**Issue by martinratinaud** [OpenTermsArchive/contribution-tool#90]
There is a configuration that causes a problem on the contribution tool

- add a document to an unknown service -> creates a PR1
- add a document to the same service which does not yet exist in main -> creates a PR2
- merge PR1
- PR2 should be rebased on main when submitting a new change through the contribution tool.

I've begun working on this but unfortunately could not find a way to rebase the branch on main easily through the Octokit API

Maybe @MattiSG you can help me on that?

Thanks

**Comment by MattiSG** [comment]
No, I don't know of a any way to do such a conflict resolution automatically. I believe that if such a thing existed, it would already be a standard algorithm 😉

The one thing we can do is use business knowledge instead of line-by-line resolution, by loading both JSON files and merging them before producing the resulting PR. However, that would not prevent conflicts in the scenario you describe, since we want both PRs to be independent.

I am not sure the frequency of such cases is worth investing in solving them.

**Comment by martinratinaud** [comment]
> The one thing we can do is use business knowledge instead of line-by-line resolution, by loading both JSON files and merging them before producing the resulting PR

This is what I already do

And for the frequency, it happened already to me 5 times, taking each time between 5 and 10 minutes. So let's see if we need this in the future and reference here every PR we have a problem with
